src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -1102,7 +1102,7 @@ protected void AsyncMethodChecks(bool verifyReturnType, Location errorLocation,
 
                 if (this.HasAsyncMethodBuilderAttribute(out _))
                 {
-                    hasErrors |= MessageID.IDS_AsyncMethodBuilderOverride.CheckFeatureAvailability(diagnostics, this.DeclaringCompilation, errorLocation);
+                    hasErrors |= !MessageID.IDS_AsyncMethodBuilderOverride.CheckFeatureAvailability(diagnostics, this.DeclaringCompilation, errorLocation);
```

Comment:
>!MessageID.IDS_AsyncMethodBuilderOverride.CheckFeatureAvailability(diagnostics, this.DeclaringCompilation, errorLocation); [](http://example.com/codeflow?start=33&length=122)

In my opinion registering language version check in ```hasErrors``` is not the right thing to do. Wrong language version shouldn't change the analysis, and here, it clearly does.

Reply:
In that case, should https://github.com/dotnet/roslyn/blob/9c04fee59aae2c7faa8b02cf08bb3bcc9d637664/src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs#L226-L230 be tracked in a new issue to update as well?

Reply:
@Youssef1313 That seems a good idea. There's at least one check there that is conditional on `hasErrors`.

Reply:
Filed https://github.com/dotnet/roslyn/issues/65363